dbt/adapters/impala/connections.py

```diff
@@ -159,27 +160,51 @@ def open(cls, connection):
                     port=credentials.port,
                     retries=credentials.retries,
                 )
+            connection_end_time = time.time()
 
-            connection.state = "open"
+            connection.state = ConnectionState.OPEN
             connection.handle = handle
         except Exception as ex:
             LOGGER.debug("Connection error {}".format(ex))
-            connection.state = "fail"
+            connection.state = ConnectionState.FAIL
             connection.handle = None
+            connection_end_time = time.time()
             pass
 
         # track usage
         payload = {
             "id": "dbt_impala_open",
-            "unique_hash": hashlib.md5(credentials.host.encode()).hexdigest(),
+            "unique_hash": hashlib.md5((str(credentials.host) + str(credentials.username)).encode()).hexdigest(),
             "auth": auth_type,
+            "connection_name": connection.name,
             "connection_state": connection.state,
+            "connection_opening_time": connection_end_time - connection_start_time,
         }
 
         tracker.track_usage(payload)
 
         return connection
 
+    @classmethod
+    def close(cls, connection):
+        try:
+            connection_start_time = time.time()
+            super().close(connection)
+            connection_end_time = time.time()
+
+            credentials = connection.credentials
+
+            payload = {
+                "id": "dbt_impala_close",
+                "unique_hash": hashlib.md5((str(credentials.host) + str(credentials.username)).encode()).hexdigest(),
+                "connection_name": connection.name,
+                "connection_closing_time": connection_end_time - connection_start_time,
```

Comment:
We need this time in a human readable format preferably utc.

Reply:
This is elapsed time, not timestamp,

Reply:
Ok then we need to rename this variable.

Reply:
Changed to:

ganesh.venkateshwara@MacBook-Pro dbtdemo % grep "Tracker adapter:" logs/dbt.log
15:54:21.216147 [debug] [MainThread]: Tracker adapter: Usage tracking flag True
15:54:21.217620 [debug] [Thread-1  ]: Tracker adapter: Sending Event [{"data": {"id": "dbt_impala_open", "unique_hash": "0d957f2dd9707f9eea2aa91a24858cb4", "auth": "insecure", "connection_name": "debug", "connection_state": "open", "connection_opening_time": "0.01"}}]
15:54:21.245038 [debug] [MainThread]: Tracker adapter: Usage tracking flag True
15:54:21.245470 [debug] [Thread-2  ]: Tracker adapter: Sending Event [{"data": {"id": "dbt_impala_close", "unique_hash": "0d957f2dd9707f9eea2aa91a24858cb4", "connection_name": "debug", "connection_state": "closed", "connection_closing_time": "0.00"}}]